doc/adrs/158_add_support_for_styled_labels_in_view_model_for_trees.adoc

```diff
@@ -0,0 +1,104 @@
+This is a sort of catalog of all defined style that could be used (or not) for a particular tree item label (or one of its part).
+We do prefer wrap all text styles in a dedicated property to avoid to mix them with other part of the tree specification (such as menu actions in a further work).
+A similar approach has been followed for storing colors in a palette inside diagrams specification.
+This property will be automatically created for each tree specification.
+
+The second owned property (``labelStyles``) is the actual set of styles that are used for displaying labels in this tree specification.
+
+```
+StylePalette {
+  textStyles : TextStyleDescription [0..*]
+}
+
+TextStyleDescription {
+  name: Identifier
+  fontNameExpression: InterpretedExpression<string>
+  foregroundColorExpression: InterpretedExpression<string>
+  backgroundColorExpression: InterpretedExpression<string>
+  isUnderlineExpression: InterpretedExpression<boolean>
+  isBoldExpression: InterpretedExpression<boolean>
+  isItalicExpression: InterpretedExpression<boolean>
+}
+```
+
+The above style description allows to specify a set of text style details.
+Such a description is named for clarity sake and is referenced inside conditional styled labels via their fragments.
+The ``fontNameExpression`` should return the name of the font to use.
+The color of the label fragment is specified through both ``foregroundColorExpression`` and ``backgroundColorExpression``.
+The string returned by those expressions should be compliant with a CSS color (e.g. ``#rrggbb``).
+The ``isUnderlineExpression`` should return whether this fragment is underlined or not.
+By design, we narrow the expressivity of the underline style to a solid line with the same color as expressed by the ``foregroundColorExpression`` expression.
+The ``isBoldExpression`` and ``isItalicExpression`` expressions respectively control the bold and italic style of this fragment.
+Those three last expressions should be evaluated in a boolean.
+
+```
+ConditionalStyledLabel {
+  name: Identifier
+  conditionExpression: InterpretedExpression<boolean>
+  fragments: StyledLabelFragment [1..*]
+}
+
+StyledLabelFragment {
+  labelExpression: InterpretedExpression<string>
+  style: TextStyleDescription (reference)
+}
+```
+
+The styles applied to labels (or portions of label) are named and conditional and they hold an ordered list of styled label fragments.
+In case the evaluation of its ``conditionExpression`` returns ``false`` no one of embedded fragments will be considered and an unstyled label will be used using the ``treeItemLabelExpression`` result.
+If ``conditionExpression`` returns ``true``, fragments are considered in the specified order.
+Each of them specifies the style of a sub-part of the label. 
+The ``labelExpression`` of the fragment should return the string itself and the style to apply is given through the reference of an actual TextStyleDescription through the ``style`` property.
```

Comment:
I totally agree with you solution for complex labels. But initial the Form DSL dit not provide the dynamic mapping at the begining so this solution would have been a first step (and sufficient for our use case). Afteward, we could have added the dynamic item to express the if and loop feature. 

I agree that the complex example should be taken in consideration now to avoid any API break in the future. So you examples are good to keep in mind but it was not our intention to handle them as a first step.

Do you think that a first stage that do not handle complexe example as your is acceptable, or should we implement it now? (I'm affraid for my budget)

As a first step, we could introduce an abstract class that would allow us to extend with dynamic concepts afterwards?

Moreover, in our solution we add the capacity to express a meaninfull style. For example, as a specifier I want to create a _Read Only_ style that display a text in _italic_ and grey (or any another combination). Then I want to be able to reuse that style in many different scenario. The fact to link the the precondition, labe expression and the definition of the style make them not really reusable. It would mean for the specifier to dupplicate many time the _colorExpression_, _isItalicExpresion_ and so on.

For example, in your example the "Black" style is repeated a lot of time. With our solution we would have a TextStyleDescription name "Default Style" with difine this "Black Color". Then it would be referenced any time needed.